<?xml version="1.0"?>
<robot name="threeseg">
  <link name="base" />
  <link name="link1">
    <visual>
      <origin xyz="0 0 1.5" rpy="0 0 0"/>
      <geometry>
        <box size="0.1 0.1 3"/>
      </geometry>
    </visual>
  </link>
  <link name="link2">
    <visual>
      <origin xyz="0 0 1" rpy="0 0 0"/>
      <geometry>
        <box size="0.1 0.1 2"/>
      </geometry>
    </visual>
  </link>
  <link name="link3">
    <visual>
      <origin xyz="0 0 0.5" rpy="0 0 0"/>
      <geometry>
        <box size="0.1 0.1 1"/>
      </geometry>
    </visual>
  </link>
  <link name="tool"></link>

  <joint name="joint1" type="revolute">
    <parent link="base" />
    <child link="link1" />
    <limit effort="30" velocity="1.0" lower="-1.570795" upper="1.570795" />
    <origin xyz="0 0 0" rpy="0 0 0" />
    <axis xyz="0 0 1" />
  </joint>
  <joint name="joint2" type="revolute">
    <parent link="link1" />
    <child link="link2" />
    <limit effort="30" velocity="1.0" lower="-1.570795" upper="1.570795" />
    <origin xyz="0 0 3" rpy="-0.785398 0 0" />
    <axis xyz="1 0 0" />
  </joint>
  <joint name="joint3" type="revolute">
    <parent link="link2" />
    <child link="link3" />
    <limit effort="30" velocity="1.0" lower="-1.570795" upper="1.570795" />
    <origin xyz="0 0 2" rpy="-0.5 0 0" />
    <axis xyz="1 0 0" />
  </joint>
  <joint name="ee" type="fixed">
    <parent link="link3" />
    <child link="tool" />
    <limit effort="30" velocity="1.0" lower="-6.28318" upper="6.28318" />
    <origin xyz="0 0 1" rpy="0 0 0" />
    <axis xyz="0 0 1" />
  </joint>
</robot>